dist/update-release-notes.go

```diff
@@ -0,0 +1,449 @@
+	pathToChangeLog := getChangeLogPath()
+	file, err := os.Open(pathToChangeLog)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	changeLog := getChangeLogInfo(bufio.NewScanner(file))
+	releaseNotes := createReleaseNotes(changeLog)
+	tags := getTagsFromGitHub(*tagsRepoPtr)
+	repoPath := *repoPtr
+	releaseTagValidationInput := releaseTagValidationInput{
+		authToken:    *authTokenPtr,
+		tags:         tags,
+		changeLog:    changeLog,
+		releaseNotes: releaseNotes,
+		repoPath:     repoPath,
+	}
+	releaseMetaData := validateTagsHaveReleases(releaseTagValidationInput)
+	createMissingReleases(releaseMetaData, *authTokenPtr, repoPath)
+	fmt.Println("Created missing releases.")
+	updateReleasesIfChanged(releaseMetaData, *authTokenPtr, repoPath)
+	fmt.Println("Updated releases.")
+}
+
+func parseFlags() (*bool, *string, *string, *string) {
+	help := flag.Bool("help", false, "Print usage information. Example: $ go run update-release-notes.go -Repo=quick-lint/quick-lint-js -TagsRepo=quick-lint/quick-lint-js -AuthToken=$(cat token.txt)")
+	authTokenPtr := flag.String("AuthToken", "", "Visit: (https://github.com/settings/tokens) generate a token with 'public_repo' or 'repo' permissions. Store access token in a file (token.txt). Example usage: -AuthToken=$(cat token.txt)")
+	repoPtr := flag.String("Repo", "quick-lint/quick-lint-js", "GitHub repo where release notes to be released.")
+	tagsRepoPtr := flag.String("TagsRepo", "quick-lint/quick-lint-js", "GitHub repo to get release tags from.")
+	flag.Parse()
+	if *authTokenPtr == "" {
+		fmt.Println(redColor + "WARNING: No GitHub access token given for flag -AuthToken. Refer to --help" + resetColor)
+	}
+	return help, authTokenPtr, repoPtr, tagsRepoPtr
+}
+
+func displayHelp(help *bool) {
+	if *help {
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTIONS]\n", os.Args[0])
+		flag.PrintDefaults()
+		os.Exit(0)
+	}
+}
+
+func getChangeLogPath() string {
+	_, filename, _, _ := runtime.Caller(0)
+	pathToChangeLog := filepath.Join(filepath.Dir(filename), "../docs/CHANGELOG.md")
+	return pathToChangeLog
+}
+
+func handleChannels(channel chan bool, waitGroup *sync.WaitGroup) {
+	timeout := time.After(10 * time.Second)
+	for {
+		select {
+		case <-channel:
+			if len(channel) == 0 {
+				waitGroup.Wait()
+				close(channel)
+				return
+			}
+		case <-timeout:
+			fmt.Println("Timeout: Some goroutines did not complete")
+			close(channel)
+			return
+		}
+	}
+}
+
+func createMissingReleases(releaseMetaData releaseMetaData, authToken string, repoPath string) {
+	createReleaseWaitGroup := &sync.WaitGroup{}
+	createReleaseChannel := make(chan bool, len(releaseMetaData.ReleaseVersionTagMap))
+
+	spawnedThread := false
+	makeLatestRelease := "false"
+	for releaseVersion, tagVersion := range releaseMetaData.ReleaseVersionTagMap {
+		if releaseMetaData.ReleaseVersionTagMap[releaseVersion] == releaseMetaData.TagReleaseVersionMap[tagVersion] {
+			if releaseVersion == releaseMetaData.LatestReleaseVersion {
```

Comment:
Nit: Make the `makeLatestRelease` variable as local as possible.
```suggestion
	for releaseVersion, tagVersion := range releaseMetaData.ReleaseVersionTagMap {
		if releaseMetaData.ReleaseVersionTagMap[releaseVersion] == releaseMetaData.TagReleaseVersionMap[tagVersion] {
			var makeLatestRelease string
			if releaseVersion == releaseMetaData.LatestReleaseVersion {
```